src/components/posts/index.js

```diff
@@ -1,27 +1,49 @@
 import { useEffect, useState } from "react";
 import Post from "../post";
-import { getPosts } from "../../service/apiClient";
 
 const Posts = () => {
-    const [posts, setPosts] = useState([])
+  const [posts, setPosts] = useState([]);
 
-    useEffect(() => {
-        getPosts().then(setPosts)
-    }, [])
+  useEffect(() => {
+    const token = localStorage.getItem('token'); // Adjust according to where your token is stored
```

Comment:
Let's change this `'token'` to a const so it is easier to change later. Let's also remove the comment